crates/mdb/src/error.rs

```diff
@@ -73,3 +58,22 @@ impl Error {
         unsafe { pchar_to_string(self.as_ref().message) }
     }
 }
+
+impl std::ops::Deref for BorrowedError<'_> {
+    type Target = Error;
+    // SAFETY: self.ptr is effectively a reference to Error provided by mdb-sys for the duration
+    // of a callback. BorrowedError is tied to the lifetime of the contained pointer and since
+    // Error does not implement Copy there is no way to leak the contained pointer.
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*(self.ptr as *const Error) }
+    }
+}
+
+impl BorrowedError<'_> {
+    pub(crate) fn new(ptr: *const mdb_sys::mdb_error_t) -> Self {
```

Comment:
I think this should be unsafe because there are preconditions that the caller must ensure for this function to be sound. I know it kept the safe marking from `new_borrowed`, so you don't have to change it now. But we should change it eventually.